### ISSUE TITLE
Git pull fail for 0.7.0.8 to 0.7.0.9 on a todays fresh install and redirect issues

### ISSUE THREAD

**Issue by Roversword** [TechnicPack/TechnicSolder#282]
Hi all
I have two issues with technicsolder at the moment

I was using the following guides:
https://github.com/TechnicPack/TechnicSolder/wiki/Getting-Started
https://github.com/TechnicPack/TechnicSolder/wiki/How-To-Install-TechnicSolder-for-Linux-%7BWIP%7D
and consequently:
https://www.digitalocean.com/community/tutorials/how-to-install-linux-nginx-mysql-php-lemp-stack-on-ubuntu-14-04

I am using apache2.4 on Ubuntu 14.04
I do NOT use mysql, using sqlite (default)

The following was done 2015-01-27 (today), which should result in having TechnicSolder 0.7.0.9 and all updates from Ubuntu at this time.

installed software:
sudo apt-get install php5 php5-cli php5-curl php5-gd php5-mcrypt php5-mysql curl git apache2 ngnix php5-sqlite libapache2-mod-php5

Made a folder on my datadisk and gave it to myuser:www-data
/datadisk/technicsolder

Within this folder I put the composer:
curl -sS https://getcomposer.org/installer | php
Checked if there is an update: sudo php composer.phar self-update (there wasn't at the time)

As mentioned in the manual I downloaded from GIT, moved it in my working directory and then run the composer command:
git clone https://github.com/TechnicPack/TechnicSolder.git
moved it to /datadisk/technicsolder
then (in /datadisk/technicsolder)
php composer.phar install --no-interaction

after that I made a chown -R myuser:www-data over all files (even though only app and public would need it, I guess)

after that I did the migration:
php artisan migrate:install (which failed, because there was already a table "migration")
php artisan migrate (which ended in "nothing to migrate")

Make a "mods"-directory in /datadisk/technicsolder/public/modpack/mods/
Same permissions as the rest (myuser:www-data, 775)

Then changed the values in solder.php as mentioned'
repo_location' => '/datadisk/technicsolder/modpack/mods/'

Pointed apache (root directory) to /datadisk/technicsolder/public/

Now that technically should make me able to surf to my website
Apache has received /datadisk/technicsolder/public/ (with trailing slash) as root directory.
I put a info.php into the public folder so I could check php functionality (which worked) 

When I then to go mysite.domain.ch (which points to /datadisk/technicsolder/public/) I receive a 404, because I get directly redirected to mysite.domain.ch/dashboard.
As I understand that is not okay, right?
I then manually go to mysite.domain.ch/index.php which offers me a login, which ended in an Error in exception handler.

I then deleted the app/database/production.sqlite, made a new file (touch), gave it a 775 and myuser:www-data...and made a php artisan migrate:install and migrate (which actuall did stuff this time)

The problem with the dashboard still persisted, so I went to index.php manually with the browser
I was logged in this time (automagically, but it worked) and was offered 0.7.0.8 (not .9) of technicsolder.

After changing the credentials I logged out and in again, and then was told there is an update available.

So I went back to ssh and did what the wiki told me (all in /datadisk/technicsolder):
php artisan down
git pull -> failed, because it said that I needed to move stuff (in app and public) and aborted then.
-> there is quite a list of files which would need moving, do I really need to do that every time?

May someone assist me in finding out what I am doing wrong? (the redirect and the fact I am still on 0.7.0.8 and the git pull failed)

Thanks a lot!

**Comment by GenPage** [comment]
Hey there, try doing a `git reset --hard v0.7.0.8` And then `git pull`. The system thinks local files were changed and doesn't want to override them

**Comment by Roversword** [comment]
Hi GenPage

Thanks a lot.
I tried, however, it appears that some files are still considered an issue:

git reset --hard v0.7.0.8:
Updating 5dfbd1e..203399egit reset --hard v0.7.0.8
HEAD is now at 5dfbd1e Update to MinecraftUtils for better timeout handling, Changed User Agent to remove spoofing.

git pull:
error: The following untracked working tree files would be overwritten by merge:
	app/views/errors/500.blade.php
	app/views/errors/503.blade.php
	app/views/errors/missing.blade.php
	public/css/errors.css
	public/css/solder.css
	public/img/500error2.png
	public/img/background.png
	public/img/title.png
Please move or remove them before you can merge.
Aborting

Thanks

**Comment by GenPage** [comment]
Interesting, those files should of been removed with a "--hard" reset. You can manually remove using `rm` and then try again

Example: `rm app/views/errors/500.blade.php`

**Comment by Roversword** [comment]
Hi GenPage

I manually removed all the files mentioned above (which git wanted removed) and tried another git pull...this time it worked.
the php composer.phar update worked as well.
I logged in afterwards and was greeted by 0.7.0.9. So it worked
Still, the questions resides as why it happend

As for the redirect, I did what was proposed to me in http://forums.technicpack.net/topic/111376-solder-where-is-the-dashboard/
I changed the app/routes.php and that appears to be working. Maybe something to include in 0.7.0.10?
